Add Training on VOC dataset sample codes.

genausz opened the pull request:
Hello,
I implemented Pascal VOC Segmentation task on Mask RCNN.
Check it out.

genausz committed:
trainning on VOC

genausz committed:
inference

genausz committed:
transfer learning from coco weight

genausz committed:
transfer training from coco weight

genausz committed:
added training from coco comments

genausz committed:
Merge branch 'master' of https://github.com/genausz/Mask_RCNN

genausz committed:
fix image save suffix

tobenan commented:
@genausz ,hi, how do you evaluate the quailty of the mask?

SalahAdDin commented:
@genausz What's about you don't have any Segmentation file and others? Just training, validation, and testing files.

SalahAdDin commented:
Also, what's about using **ResNet** with Mask R-CNN?

genausz commented:
> @genausz What's about you don't have any Segmentation file and others? Just training, validation, and testing files.

Hello, you can download VOC dataset from web( like https://pjreddie.com/projects/pascal-voc-dataset-mirror/), and set the right path run trainning, validation testing process.

SalahAdDin commented:
@genausz My problem is i have custom datasets just with main sets, nor segmentation set nor others.

None of them has **Segmentation** files, just main files, annotations, and normal images, in that case, how can I use this pull request? I haven't problems with other frameworks, but I require to use this framework too. 

Does `Mask R-CNN`  require the mask files?

SalahAdDin commented:
@genausz I made this [pull request](https://github.com/matterport/Mask_RCNN/pull/2006) with some changes in order to enable different VOC formatted datasets.

genausz commented:
> @genausz I made this [pull request](https://github.com/matterport/Mask_RCNN/pull/2006) with some changes in order to enable different VOC formatted datasets.

Great job.

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 26,061.91kb -> 24,530.70kb (5.88%)

/assets/detection_histograms.png -- 13.43kb -> 6.23kb (53.62%)
/assets/detection_tensorboard.png -- 43.08kb -> 28.14kb (34.69%)
/assets/mapping_challenge.png -- 613.20kb -> 450.15kb (26.59%)
/assets/project_grass_gis.png -- 578.81kb -> 499.38kb (13.72%)
/assets/project_ice_wedge_polygons.png -- 1,028.26kb -> 909.96kb (11.51%)
/assets/project_3dbuildings.png -- 959.02kb -> 849.53kb (11.42%)
/assets/project_shiny1.jpg -- 216.48kb -> 194.04kb (10.37%)
/assets/street.png -- 890.48kb -> 798.17kb (10.37%)
/images/8734543718_37f6b8bd45_z.jpg -- 162.83kb -> 146.90kb (9.79%)
/assets/project_usiigaci1.gif -- 2,597.78kb -> 2,428.11kb (6.53%)
/images/3132016470_c27baa00e8_z.jpg -- 169.62kb -> 158.71kb (6.43%)
/images/9247489789_132c0d534a_z.jpg -- 282.17kb -> 264.81kb (6.15%)
/images/7933423348_c30bd9bd4e_z.jpg -- 208.56kb -> 196.33kb (5.86%)
/images/6821351586_59aa0dc110_z.jpg -- 212.46kb -> 200.51kb (5.63%)
/images/2383514521_1fc8d7b0de_z.jpg -- 203.23kb -> 192.29kb (5.38%)
/images/6584515005_fce9cec486_z.jpg -- 142.24kb -> 134.61kb (5.36%)
/images/8433365521_9252889f9a_z.jpg -- 177.85kb -> 168.33kb (5.35%)
/images/3627527276_6fe8cd9bfe_z.jpg -- 157.16kb -> 148.85kb (5.29%)
/images/2516944023_d00345997d_z.jpg -- 176.98kb -> 167.64kb (5.27%)
/images/8829708882_48f263491e_z.jpg -- 237.28kb -> 225.05kb (5.15%)
/images/25691390_f9944f61b5_z.jpg -- 176.20kb -> 167.40kb (4.99%)
/assets/balloon_color_splash.gif -- 9,426.26kb -> 8,962.65kb (4.92%)
/images/8699757338_c3941051b6_z.jpg -- 234.31kb -> 223.06kb (4.8%)
/images/1045023827_4ec3e8ba5c_z.jpg -- 137.64kb -> 131.08kb (4.77%)
/images/9118579087_f9ffa19e63_z.jpg -- 300.74kb -> 286.64kb (4.69%)
/images/8053677163_d4c8f416be_z.jpg -- 219.62kb -> 209.36kb (4.67%)
/images/4782628554_668bc31826_z.jpg -- 223.71kb -> 213.30kb (4.65%)
/images/8239308689_efa6c11b08_z.jpg -- 221.09kb -> 210.84kb (4.64%)
/images/262985539_1709e54576_z.jpg -- 121.88kb -> 116.60kb (4.33%)
/images/7581246086_cf7bbb7255_z.jpg -- 142.37kb -> 136.30kb (4.26%)
/images/3651581213_f81963d1dd_z.jpg -- 147.02kb -> 140.81kb (4.22%)
/images/3878153025_8fde829928_z.jpg -- 85.81kb -> 82.27kb (4.13%)
/images/3800883468_12af3c0b50_z.jpg -- 119.68kb -> 114.82kb (4.06%)
/images/5951960966_d4e1cda5d0_z.jpg -- 157.21kb -> 151.40kb (3.7%)
/images/8512296263_5fc5458e20_z.jpg -- 145.92kb -> 140.59kb (3.65%)
/images/2502287818_41e4b0c4fb_z.jpg -- 130.58kb -> 126.26kb (3.31%)
/images/3862500489_6fd195d183_z.jpg -- 94.01kb -> 91.30kb (2.88%)
/assets/detection_masks.png -- 9.75kb -> 9.47kb (2.82%)
/images/4410436637_7b0ca36ee7_z.jpg -- 123.79kb -> 120.47kb (2.68%)
/assets/detection_final.png -- 887.11kb -> 871.45kb (1.77%)
/assets/detection_activations.png -- 69.11kb -> 67.92kb (1.73%)
/assets/detection_refinement.png -- 702.75kb -> 694.57kb (1.16%)
/images/12283150_12d37e6389_z.jpg -- 66.75kb -> 66.00kb (1.13%)
/assets/detection_anchors.png -- 746.79kb -> 739.00kb (1.04%)
/assets/images_to_osm.png -- 983.02kb -> 974.56kb (0.86%)
/assets/nucleus_segmentation.png -- 708.45kb -> 706.25kb (0.31%)
/assets/project_usiigaci2.gif -- 609.42kb -> 608.60kb (0.13%)

Signed-off-by: ImgBotApp <ImgBotHelp@gmail.com>

genausz committed:
Merge pull request #1 from genausz/imgbot

[ImgBot] Optimize images
